src/main/api/ledger.ts

```diff
@@ -102,23 +108,91 @@ const signBtcTxInLedger = async (
       segwit: true,
       additionals: ['bitcoin', 'bech32']
     })
-    await transport.close()
 
     return E.right(txHex)
   } catch (error) {
-    return E.left(getErrorId(error.message, error.statusText))
+    return E.left(getErrorId(error.toString()))
   }
 }
 
-export const signTxInLedger = (
+const sendBNCTxInLedger = async (
+  transport: TransportNodeHid,
+  network: Network,
+  ledgerTxInfo: LedgerBNCTxInfo
+): Promise<E.Either<LedgerErrorId, string>> => {
+  try {
+    const { sender, recipient, asset, amount, memo } = ledgerTxInfo
+    const client = new XChainBinanceClient({ network: network === 'testnet' ? 'testnet' : 'mainnet' })
+    const ledgerApp = new LedgerAppBNB(transport)
+    const derive_path = getBNBDerivePath(0)
+    const hpr = network === 'testnet' ? 'tbnb' : 'bnb' // This will be replaced later with "const hpr = client.getPrefix()"
+    await ledgerApp.showAddress(hpr, derive_path)
+
+    const bncClient = client.getBncClient()
+    bncClient.initChain()
+
+    bncClient.useLedgerSigningDelegate(
+      ledgerApp,
+      () => {},
+      () => {},
+      () => {},
+      derive_path
+    )
+
+    const transferResult = await bncClient.transfer(
+      sender,
+      recipient,
+      baseToAsset(amount).amount().toString(),
+      asset ? asset.symbol : AssetBNB.symbol,
+      memo
+    )
+
+    return E.right(transferResult.result.map((txResult: { hash?: TxHash }) => txResult?.hash ?? '')[0])
+  } catch (error) {
+    return E.left(getErrorId(error.toString()))
+  }
+}
+
+export const signTxInLedger = async (
+  chain: Chain,
+  network: Network,
+  ledgerTxInfo: LedgerTxInfo
+): Promise<E.Either<LedgerErrorId, string>> => {
+  try {
+    const transport = await TransportNodeHid.open('')
+    let res
+    switch (chain) {
+      case 'BTC':
+        res = await signBTCTxInLedger(transport, network, ledgerTxInfo as LedgerBTCTxInfo)
+        break
+      default:
+        res = E.left(LedgerErrorId.NO_APP)
+    }
+    await transport.close()
+    return res
+  } catch (error) {
+    return E.left(getErrorId(error.toString()))
+  }
+}
+
+export const sendTxInLedger = async (
   chain: Chain,
   network: Network,
   ledgerTxInfo: LedgerTxInfo
 ): Promise<E.Either<LedgerErrorId, string>> => {
-  switch (chain) {
-    case 'BTC':
-      return signBtcTxInLedger(network, ledgerTxInfo)
-    default:
-      return Promise.reject(E.left(LedgerErrorId.NO_APP))
+  try {
+    const transport = await TransportNodeHid.open('')
+    let res
+    switch (chain) {
+      case 'BNB':
+        res = await sendBNCTxInLedger(transport, network, ledgerTxInfo as LedgerBNCTxInfo)
+        break
+      default:
+        res = E.left(LedgerErrorId.NO_APP)
+    }
+    await transport.close()
+    return res
+  } catch (error) {
+    return E.left(getErrorId(error.toString()))
```

Comment:
i still suggesting to decompose this file by chain/operation meanings. it becomes to the 200 lines already and as i understand this is not even a half f functionality we will have for ledger

Reply:
I agree but can we do this later in another PR? What do you think? @veado

Reply:
np - just create an issue for it and do it in another PR.

Reply:
https://github.com/thorchain/asgardex-electron/issues/676